--- WEB-INF/classes/hudson/plugins/mstest/mstest-to-junit.xsl.orig	Thu Feb 11 20:26:00 2010
+++ WEB-INF/classes/hudson/plugins/mstest/mstest-to-junit.xsl	Mon Nov 22 13:07:21 2010
@@ -44,8 +44,21 @@
 				</xsl:for-each>
 				
 				<xsl:for-each select="//b:UnitTestResult">
-					<xsl:variable name="testName" select="@testName"/>
-					<xsl:variable name="executionId" select="@executionId"/>
+					<xsl:variable name="testName">
+					<xsl:value-of select="@testName"/>
+						<xsl:if test="@dataRowInfo"> row <xsl:value-of select="@dataRowInfo"/>
+						</xsl:if>
+					</xsl:variable>
+					<xsl:variable name="executionId">
+						<xsl:choose>
+							<xsl:when test="@resultType = 'DataDrivenDataRow'">
+								<xsl:value-of select="@parentExecutionId"/>
+							</xsl:when>
+							<xsl:otherwise>
+								<xsl:value-of select="@executionId"/>
+							</xsl:otherwise>
+						</xsl:choose>
+					</xsl:variable>
 					<xsl:variable name="duration_seconds" select="substring(@duration, 7)"/>
 					<xsl:variable name="duration_minutes" select="substring(@duration, 4,2 )"/>	
 					<xsl:variable name="duration_hours" select="substring(@duration, 1, 2)"/>
