Index: src/main/java/hudson/plugins/dependencyanalyzer/parser/BuildLogFileParser.java
===================================================================
--- src/main/java/hudson/plugins/dependencyanalyzer/parser/BuildLogFileParser.java	(Revision 20959)
+++ src/main/java/hudson/plugins/dependencyanalyzer/parser/BuildLogFileParser.java	(Arbeitskopie)
@@ -33,7 +33,7 @@
 			+ ".*");
 
 	private enum Goal {
-		DEPENDENCY_ANALYSE(LOG_LEVEL_REGEX + "\\[dependency:analyze\\]$");
+		DEPENDENCY_ANALYSE(LOG_LEVEL_REGEX + "\\[dependency:analyze(-only)?\\]$");
 
 		private Pattern pattern;
 
