Index: plugins/rubyMetrics/src/main/java/hudson/plugins/rubyMetrics/rcov/model/MetricTarget.java
===================================================================
--- plugins/rubyMetrics/src/main/java/hudson/plugins/rubyMetrics/rcov/model/MetricTarget.java	(revision 33501)
+++ plugins/rubyMetrics/src/main/java/hudson/plugins/rubyMetrics/rcov/model/MetricTarget.java	(working copy)
@@ -1,6 +1,7 @@
 package hudson.plugins.rubyMetrics.rcov.model;
 
 import org.apache.commons.beanutils.Converter;
+import org.kohsuke.stapler.DataBoundConstructor;
 
 public class MetricTarget {
 	
@@ -18,6 +19,7 @@
      * @param unstable
      * @stapler-constructor
      */
+    @DataBoundConstructor
 	public MetricTarget(Targets metric, Integer healthy, Integer unhealthy, Integer unstable) {		
 		this.metric = metric;
 		this.healthy = healthy != null?healthy:80;
