Index: src/main/java/hudson/plugins/warnings/parser/GccParser.java
===================================================================
--- src/main/java/hudson/plugins/warnings/parser/GccParser.java	(Revision 13311)
+++ src/main/java/hudson/plugins/warnings/parser/GccParser.java	(Arbeitskopie)
@@ -13,12 +13,12 @@
     /** Warning type of this parser. */
     static final String WARNING_TYPE = "gcc";
     /** Pattern of gcc compiler warnings. */
-    private static final String GCC_WARNING_PATTERN = "(.*\\.[chpsola0-9]+):(?:(\\d*):(?:\\d*:)*\\s*(warning|error)\\s*:|\\s*undefined reference to)(.*)";
+    private static final String GCC_WARNING_PATTERN = "^(.*\\.[chpsola0-9]+):(?:(\\d*):(?:\\d*:)*\\s*(warning|error)\\s*:|\\s*undefined reference to)(.*)$";
     /**
      * Creates a new instance of <code>GccParser</code>.
      */
     public GccParser() {
-        super(GCC_WARNING_PATTERN);
+        super(GCC_WARNING_PATTERN, true);
     }
 
     /** {@inheritDoc} */
