### Eclipse Workspace Patch 1.0
#P cobertura
Index: src/main/resources/hudson/plugins/cobertura/dashboard/CoverageTablePortlet/main.jelly
===================================================================
--- src/main/resources/hudson/plugins/cobertura/dashboard/CoverageTablePortlet/main.jelly   (revision 34262)
+++ src/main/resources/hudson/plugins/cobertura/dashboard/CoverageTablePortlet/main.jelly   (working copy)
@@ -39,6 +39,7 @@
                    <td width="10%" class="pane-header">${%Conditionals}</td>
                    <td width="10%" class="pane-header">${%Files}</td>
                    <td width="10%" class="pane-header">${%Lines}</td>
+                   <td width="10%" class="pane-header">${%Methods}</td>
                    <td width="10%" class="pane-header">${%Packages}</td>
                  </tr>
                  <tbody>
