diff --git a/src/main/java/org/jenkinsci/plugins/jobgenerator/JobGenerator.java b/src/main/java/org/jenkinsci/plugins/jobgenerator/JobGenerator.java
index 17aab18..d0448d0 100755
--- a/src/main/java/org/jenkinsci/plugins/jobgenerator/JobGenerator.java
+++ b/src/main/java/org/jenkinsci/plugins/jobgenerator/JobGenerator.java
@@ -91,6 +91,20 @@ public class JobGenerator extends Project<JobGenerator, GeneratorRun>
         super.doBuild(req, rsp, delay);
     }
 
+
+    /**
+     * See {@link #setCustomWorkspace(String)}.
+     */
+    private String customWorkspace;
+
+    public String getCustomWorkspace() {
+        return customWorkspace;
+    }
+
+    public void setCustomWorkspace(String customWorkspace) {
+        this.customWorkspace = customWorkspace;
+    } 
+
     @Override
     public boolean isParameterized() {
         // not working for now since doBuild method of jenkins core does not
diff --git a/src/main/resources/org/jenkinsci/plugins/jobgenerator/JobGenerator/configure-advanced.jelly b/src/main/resources/org/jenkinsci/plugins/jobgenerator/JobGenerator/configure-advanced.jelly
new file mode 100644
index 0000000..4b63397
--- /dev/null
+++ b/src/main/resources/org/jenkinsci/plugins/jobgenerator/JobGenerator/configure-advanced.jelly
@@ -0,0 +1,8 @@
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:p="/lib/hudson/project">
+  <f:optionalBlock name="customWorkspace" title="${%Use custom workspace}" checked="${it.customWorkspace!=null}"
+      help="/help/project-config/custom-workspace.html">
+    <f:entry title="${%Directory}">
+      <f:textbox name="customWorkspace.directory" value="${it.customWorkspace}" />
+    </f:entry>
+  </f:optionalBlock>
+</j:jelly>
