Index: main/java/hudson/plugins/clearcase/ucm/UcmMakeBaseline.java
===================================================================
--- main/java/hudson/plugins/clearcase/ucm/UcmMakeBaseline.java	(revision 13431)
+++ main/java/hudson/plugins/clearcase/ucm/UcmMakeBaseline.java	(working copy)
@@ -316,8 +316,7 @@
         ArgumentListBuilder cmd = new ArgumentListBuilder();
 
         cmd.add("unlock");
-        cmd.add("stream:");
-        cmd.add(stream);
+        cmd.add("stream:" + stream);
 
         clearToolLauncher.run(cmd.toCommandArray(), null, null, filePath);
 
@@ -337,8 +336,7 @@
         ArgumentListBuilder cmd = new ArgumentListBuilder();
 
         cmd.add("lock");
-        cmd.add("stream:");
-        cmd.add(stream);
+        cmd.add("stream:" + stream);
 
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
 
