Index: src/main/java/hudson/plugins/emailext/ExtendedEmailPublisher.java
===================================================================
--- src/main/java/hudson/plugins/emailext/ExtendedEmailPublisher.java	(revision 27609)
+++ src/main/java/hudson/plugins/emailext/ExtendedEmailPublisher.java	(working copy)
@@ -70,6 +70,8 @@
 
     public static final String CHARSET = "utf-8";
 	
+	public static final String NEWLINE = System.getProperty("line.separator");
+
 	public static void addEmailTriggerType(EmailTriggerDescriptor triggerType) throws EmailExtException {
 		if(EMAIL_TRIGGER_TYPE_MAP.containsKey(triggerType.getMailerId()))
 			throw new EmailExtException("An email trigger type with name " +
@@ -345,6 +347,10 @@
         }
         messageContentType += "; charset=" + CHARSET;
 
+        if ("text/html".equalsIgnoreCase(messageContentType)) {
+            text = text.replaceAll(NEWLINE, NEWLINE + "<br>");
+        }
+
         msg.setContent(text, messageContentType);
     }
 
