Index: trunk/plugins/subversion/src/main/java/hudson/scm/SubversionRepositoryStatus.java
===================================================================
--- trunk/plugins/subversion/src/main/java/hudson/scm/SubversionRepositoryStatus.java	(revision 25853)
+++ trunk/plugins/subversion/src/main/java/hudson/scm/SubversionRepositoryStatus.java	(working copy)
@@ -54,7 +54,8 @@
         // compute the affected paths
         Set<String> affectedPath = new HashSet<String>();
         String line;
-        while((line=new BufferedReader(req.getReader()).readLine())!=null) {
+        BufferedReader postReader = new BufferedReader(req.getReader());
+        while((line=postReader.readLine())!=null) {
             LOGGER.finer("Reading line: "+line);
             affectedPath.add(line.substring(4));
         }
