Index: src/main/java/hudson/model/Run.java
===================================================================
--- src/main/java/hudson/model/Run.java	(revision 19066)
+++ src/main/java/hudson/model/Run.java	(working copy)
@@ -580,6 +580,15 @@
     public String getId() {
         return ID_FORMATTER.get().format(new Date(timestamp));
     }
+    
+    /**
+     * Get the date formatter used to convert the directory name in to a timestamp
+     * This is nasty exposure of private data, but needed all the time the directory
+	 * containing the build is used as it's timestamp.
+     */
+    public static SimpleDateFormat getIDFormatter() {
+    	return ID_FORMATTER.get();
+    }
 
     public Descriptor getDescriptorByName(String className) {
         return Hudson.getInstance().getDescriptorByName(className);
