diff --git a/core/src/main/resources/hudson/security/GlobalMatrixAuthorizationStrategy/config.jelly b/core/src/main/resources/hudson/security/GlobalMatrixAuthorizationStrategy/config.jelly
index e587392..5315947 100644
--- a/core/src/main/resources/hudson/security/GlobalMatrixAuthorizationStrategy/config.jelly
+++ b/core/src/main/resources/hudson/security/GlobalMatrixAuthorizationStrategy/config.jelly
@@ -151,7 +151,7 @@ THE SOFTWARE.
           e = null; <!-- avoid memory leak -->
         },
         <j:if test="${empty(descriptorPath)}">
-          <j:set var="descriptorPath" value="${rootURL}${descriptor.descriptorFullUrl}"/>
+          <j:set var="descriptorPath" value="${descriptor.descriptorFullUrl}"/>
         </j:if>
         <!-- validates the name -->
         "#${strategyid} TR.permission-row" : function(e) {
