Index: src/main/java/hudson/plugins/cigame/rules/plugins/opentasks/OpenTasksRuleSet.java
===================================================================
--- src/main/java/hudson/plugins/cigame/rules/plugins/opentasks/OpenTasksRuleSet.java	(revision 24406)
+++ src/main/java/hudson/plugins/cigame/rules/plugins/opentasks/OpenTasksRuleSet.java	(working copy)
@@ -1,7 +1,7 @@
 package hudson.plugins.cigame.rules.plugins.opentasks;
 
 import hudson.plugins.cigame.rules.plugins.PluginRuleSet;
-import hudson.plugins.tasks.util.model.Priority;
+import hudson.plugins.analysis.util.model.Priority;
 
 public class OpenTasksRuleSet extends PluginRuleSet {
 
