Index: src/main/java/hudson/plugins/emailext/EmailExtensionPlugin.java
===================================================================
--- src/main/java/hudson/plugins/emailext/EmailExtensionPlugin.java	(revision 27838)
+++ src/main/java/hudson/plugins/emailext/EmailExtensionPlugin.java	(working copy)
@@ -33,6 +33,7 @@
 		addEmailContentPlugin(new ProjectNameContent());
 		addEmailContentPlugin(new ProjectURLContent());
 		addEmailContentPlugin(new SVNRevisionContent());
+		addEmailContentPlugin(new PerforceChangeListContent());
         addEmailContentPlugin(new CauseContent());
 		
 		addEmailTriggerPlugin(PreBuildTrigger.DESCRIPTOR);
