Index: trunk/src/main/java/hudson/plugins/clover/results/PackageCoverage.java
===================================================================
--- trunk/src/main/java/hudson/plugins/clover/results/PackageCoverage.java	(revision 806)
+++ trunk/src/main/java/hudson/plugins/clover/results/PackageCoverage.java	(working copy)
@@ -66,6 +66,10 @@
         Run prevBuild = owner.getPreviousBuild();
         if (prevBuild == null) return null;
         CloverBuildAction action = prevBuild.getAction(CloverBuildAction.class);
+        while (action == null && prevBuild != null) {
+            prevBuild = prevBuild.getPreviousBuild();
+            action = prevBuild.getAction(CloverBuildAction.class);
+        }
         if (action == null) return null;
         return action.findPackageCoverage(getName());
     }
