Index: plugins/ssh-slaves/src/main/java/hudson/plugins/sshslaves/SSHLauncher.java
===================================================================
--- plugins/ssh-slaves/src/main/java/hudson/plugins/sshslaves/SSHLauncher.java	(revision 16960)
+++ plugins/ssh-slaves/src/main/java/hudson/plugins/sshslaves/SSHLauncher.java	(working copy)
@@ -519,7 +519,9 @@
             return Arrays.asList("java",
                     "/usr/bin/java",
                     "/usr/java/default/bin/java",
-                    "/usr/java/latest/bin/java");
+                    "/usr/java/latest/bin/java",
+                    "/usr/local/bin/java",
+                    "/usr/local/java/bin/java");
         }
     }
 }
