Index: src/main/java/hudson/scm/SubversionSCM.java
===================================================================
--- src/main/java/hudson/scm/SubversionSCM.java	(revision 39893)
+++ src/main/java/hudson/scm/SubversionSCM.java	(working copy)
@@ -516,14 +516,14 @@
         try {
             Map<String,Long> revisions = parseRevisionFile(build);
             if(svnLocations.length==1) {
-                Long rev = revisions.get(svnLocations[0].remote);
+                Long rev = revisions.get(getUrlWithoutRevision(svnLocations[0].remote));
                 if(rev!=null) {
                     env.put("SVN_REVISION",rev.toString());
                     env.put("SVN_URL",svnLocations[0].getURL());
                 }
             } else if(svnLocations.length>1) {
                 for(int i=0;i<svnLocations.length;i++) {
-                    Long rev = revisions.get(svnLocations[i].remote);
+                    Long rev = revisions.get(getUrlWithoutRevision(svnLocations[i].remote));
                     if(rev!=null) {
                         env.put("SVN_REVISION_"+(i+1),rev.toString());
                         env.put("SVN_URL_"+(i+1),svnLocations[i].getURL());
