From 11a09f479a1128dc11b2802fba8be564c39de566 Mon Sep 17 00:00:00 2001
From: Matt Ranostay <mranostay@gmail.com>
Date: Wed, 4 May 2011 12:15:59 -0500
Subject: [PATCH 2/2] Parameter name was misspelled in PostBuildRecorder resource

Signed-off-by: Matt Ranostay <mranostay@gmail.com>
---
 .../GroovyPostbuildRecorder/config.jelly           |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/main/resources/org/jvnet/hudson/plugins/groovypostbuild/GroovyPostbuildRecorder/config.jelly b/src/main/resources/org/jvnet/hudson/plugins/groovypostbuild/GroovyPostbuildRecorder/config.jelly
index 5ac4ece..0a7d1c2 100644
--- a/src/main/resources/org/jvnet/hudson/plugins/groovypostbuild/GroovyPostbuildRecorder/config.jelly
+++ b/src/main/resources/org/jvnet/hudson/plugins/groovypostbuild/GroovyPostbuildRecorder/config.jelly
@@ -27,7 +27,7 @@ THE SOFTWARE.
     <f:textarea field="groovyScript" />
   </f:entry>
   <f:entry title="If the script fails:" field="behavior">
-    <select name="groovypostbuild.groovyPostbuilRecorder.behavior" align="right" >
+    <select name="groovypostbuild.groovyPostbuildRecorder.behavior" align="right" >
       <f:option value="0" selected="${instance.behavior == 0}">Do nothing</f:option>
       <f:option value="1" selected="${instance.behavior == 1}">Mark build as unstable</f:option>
       <f:option value="2" selected="${instance.behavior == 2}">Mark build as failed</f:option>
-- 
1.7.0.4

