Index: src/main/java/hudson/tasks/Ant.java
===================================================================
--- src/main/java/hudson/tasks/Ant.java	(revision 11203)
+++ src/main/java/hudson/tasks/Ant.java	(working copy)
@@ -121,6 +121,10 @@
         String normalizedTarget = targets.replaceAll("[\t\r\n]+"," ");
 
         AntInstallation ai = getAnt();
+        if (!ai.getExists()) {
+        	listener.fatalError("Cannot find executable from the choosen Ant installation \"" + ai.name + "\"");
+        	return false;
+        }
         if(ai==null)
             args.add(execName);
         else
