diff --git a/src/main/resources/hudson/plugins/robot/util/robotsummary.jelly b/src/main/resources/hudson/plugins/robot/util/robotsummary.jelly
index b912d65..a5a8cd3 100644
--- a/src/main/resources/hudson/plugins/robot/util/robotsummary.jelly
+++ b/src/main/resources/hudson/plugins/robot/util/robotsummary.jelly
@@ -76,15 +76,12 @@ limitations under the License.
   	</tr>
   </table>
   <p>
-  <j:if test="${attrs.action.owner.project.parent.name != null}">
-    <j:set var="matrixprojectname" value="${attrs.action.owner.project.parent.name}/" />
-  </j:if>
-  <a href="${rootURL}/job/${matrixprojectname}${attrs.action.owner.project.name}/${attrs.action.owner.number}/${attrs.action.urlName}">&gt; Browse results</a><br />
+  <a href="${rootURL}/${attrs.action.owner.url}${attrs.action.urlName}">&gt; Browse results</a><br />
   <j:if test="${attrs.action.logFileLink != null}">
-  	<a href="${rootURL}/job/${matrixprojectname}${attrs.action.owner.project.name}/${attrs.action.owner.number}/${attrs.action.urlName}/report/${attrs.action.logFileLink}">&gt; Open ${attrs.action.logFileLink}</a><br />
+  	<a href="${rootURL}/${attrs.action.owner.url}${attrs.action.urlName}/report/${attrs.action.logFileLink}">&gt; Open ${attrs.action.logFileLink}</a><br />
   </j:if>
   <j:if test="${attrs.action.logHtmlLink != null}">
-  	<a href="${rootURL}/job/${matrixprojectname}${attrs.action.owner.project.name}/${attrs.action.owner.number}/${attrs.action.urlName}/report/${attrs.action.logHtmlLink}">&gt; Open ${attrs.action.logHtmlLink}</a><br />
+  	<a href="${rootURL}/${attrs.action.owner.url}${attrs.action.urlName}/report/${attrs.action.logHtmlLink}">&gt; Open ${attrs.action.logHtmlLink}</a><br />
   </j:if>
   </p>
 </j:jelly>
\ No newline at end of file
