diff --git a/src/main/java/hudson/plugins/mercurial/MercurialStatus.java b/src/main/java/hudson/plugins/mercurial/MercurialStatus.java
index 385b60f..5ecccf1 100644
--- a/src/main/java/hudson/plugins/mercurial/MercurialStatus.java
+++ b/src/main/java/hudson/plugins/mercurial/MercurialStatus.java
@@ -58,6 +58,8 @@ public class MercurialStatus extends AbstractModelObject implements UnprotectedR
 
             MercurialSCM hg = (MercurialSCM) scm;
             String repository = hg.getSource();
+            // Strip credentials from the repo URL
+            repository = repository.replaceFirst("://(?:.*)(?::.*)??@", "://");
             if (url.equals(repository)) urlFound = true; else continue;
             SCMTrigger trigger = project.getTrigger(SCMTrigger.class);
             if (trigger!=null) triggerFound = true; else continue;
