diff --git a/src/main/java/au/com/rayh/XCodeBuilder.java b/src/main/java/au/com/rayh/XCodeBuilder.java
index 527d109..7ddbc85 100644
--- a/src/main/java/au/com/rayh/XCodeBuilder.java
+++ b/src/main/java/au/com/rayh/XCodeBuilder.java
@@ -509,7 +509,7 @@ public class XCodeBuilder extends Builder {
             commandLine.add("-scheme");
             commandLine.add(xcodeSchema);
             xcodeReport.append(", scheme: ").append(xcodeSchema);
-        } else if (StringUtils.isEmpty(target)) {
+        } else if (StringUtils.isEmpty(target) && StringUtils.isEmpty(xcodeWorkspaceFile)) {
             commandLine.add("-alltargets");
             xcodeReport.append("target: ALL");
         } else if(interpretTargetAsRegEx != null && interpretTargetAsRegEx) {
@@ -875,4 +875,3 @@ public class XCodeBuilder extends Builder {
 	    }
     }
 }
-
