Index: main/core/src/main/resources/hudson/model/Computer/index.jelly
===================================================================
--- main/core/src/main/resources/hudson/model/Computer/index.jelly      (revision 13501)
+++ main/core/src/main/resources/hudson/model/Computer/index.jelly      (working copy)
@@ -18,6 +18,7 @@
         <img src="${imagesURL}/48x48/${it.icon}" width="48" height="48" />
         ${it.caption}
       </h1>
+      <h6> ${it.node.nodeDescription} </h6>

       <st:include from="${it.launcher}" page="main.jelly" optional="true"/>

