--- p4-plugin-master/src/main/java/org/jenkinsci/plugins/p4/PerforceScm.java	2015-06-25 11:12:05.000000000 -0500
+++ /stash/mrose/PerforceScm.java	2015-06-27 17:27:54.969292894 -0500
@@ -190,7 +190,7 @@
 
 	/**
 	 * Construct workspace from environment and then look for changes.
-	 * 
+	 *
 	 * @param envVars
 	 * @param listener
 	 * @return
@@ -370,7 +370,14 @@
 
 		if (list.isEmpty()) {
 			// No previous build, so add current
-			list.add(task.getBuildChange());
+            Object lastChange = task.getBuildChange();
+            if (lastChange != null) {
+                List<P4ChangeEntry> changes;
+                changes = task.getChangesFull(lastChange);
+                for (P4ChangeEntry c : changes) {
+                    list.add(c);
+                }
+            }
 		}
 		return list;
 	}
@@ -514,9 +521,9 @@
 	 * in the SCM class named DescriptorImpl. The Descriptor should also contain
 	 * the global configuration options as fields, just like the SCM class
 	 * contains the configurations options for a job.
-	 * 
+	 *
 	 * @author pallen
-	 * 
+	 *
 	 */
 	@Extension
 	public static class DescriptorImpl extends SCMDescriptor<PerforceScm> {
@@ -551,7 +558,7 @@
 		 * the Descriptor's fields. To persist the fields to the global
 		 * configuration XML file, the save() method must be called. Data is
 		 * defined in the global.jelly page.
-		 * 
+		 *
 		 */
 		@Override
 		public boolean configure(StaplerRequest req, JSONObject json)
@@ -562,7 +569,7 @@
 
 		/**
 		 * Credentials list, a Jelly config method for a build job.
-		 * 
+		 *
 		 * @return A list of Perforce credential items to populate the jelly
 		 *         Select list.
 		 */
@@ -635,7 +642,7 @@
 
 	/**
 	 * Helper: find the Remote/Local Computer used for build
-	 * 
+	 *
 	 * @param workspace
 	 * @return
 	 */
@@ -653,7 +660,7 @@
 
 	/**
 	 * Helper: find the Node for slave build or return current instance.
-	 * 
+	 *
 	 * @param workspace
 	 * @return
 	 */
@@ -668,7 +675,7 @@
 
 	/**
 	 * Remote execute to find hostname.
-	 * 
+	 *
 	 * @param buildWorkspace
 	 * @return
 	 */
