Index: main/java/hudson/plugins/claim/AbstractClaimBuildAction.java
===================================================================
--- main/java/hudson/plugins/claim/AbstractClaimBuildAction.java	(revision 29677)
+++ main/java/hudson/plugins/claim/AbstractClaimBuildAction.java	(working copy)
@@ -14,7 +14,10 @@
 import org.apache.commons.lang.StringUtils;
 import org.kohsuke.stapler.StaplerRequest;
 import org.kohsuke.stapler.StaplerResponse;
+import org.kohsuke.stapler.export.Exported;
+import org.kohsuke.stapler.export.ExportedBean;
 
+@ExportedBean(defaultVisibility = 2)
 public abstract class AbstractClaimBuildAction<T extends Saveable> extends TestAction implements BuildBadgeAction,
 		ProminentProjectAction {
 
@@ -59,6 +62,7 @@
 		resp.forwardToPreviousPage(req);
 	}
 
+	@Exported
 	public String getClaimedBy() {
 		return claimedBy;
 	}
@@ -67,6 +71,7 @@
 		this.claimedBy = claimedBy;
 	}
 
+	@Exported
 	public boolean isClaimed() {
 		return claimed;
 	}
@@ -109,6 +114,7 @@
 		return Hudson.getAuthentication().getName().equals("anonymous");
 	}
 
+	@Exported
 	public String getReason() {
 		return reason;
 	}
