Index: src/main/resources/org/jvnet/hudson/plugins/backup/BackupLink/index.jelly
===================================================================
--- src/main/resources/org/jvnet/hudson/plugins/backup/BackupLink/index.jelly	(revision 30447)
+++ src/main/resources/org/jvnet/hudson/plugins/backup/BackupLink/index.jelly	(working copy)
@@ -28,7 +28,7 @@
         <st:include it="${app}" page="sidepanel.jelly"/>
         <l:main-panel>
             <h1>
-                <img src="${it.iconFileName}" alt=""/>
+                <img src="${rootURL}${it.iconFileName}" alt=""/>
                 ${it.displayName}
             </h1>
 
