--- SurefireArchiver.java	2011-10-14 13:40:08.625880600 +1000
+++ SurefireArchiver.original.java	2011-10-09 15:17:26.000000000 +1000
@@ -76,8 +76,7 @@
 
     public boolean preExecute(MavenBuildProxy build, MavenProject pom, MojoInfo mojo, BuildListener listener) throws InterruptedException, IOException {
         if (isSurefireTest(mojo)) {
-            if ((!mojo.is("org.apache.maven.plugins", "maven-failsafe-plugin", "integration-test"))
-                && (!mojo.is("eviware", "maven-soapui-plugin", "test"))) {
+            if (!mojo.is("org.apache.maven.plugins", "maven-failsafe-plugin", "integration-test")) {
                 // tell surefire:test to keep going even if there was a failure,
                 // so that we can record this as yellow.
                 // note that because of the way Maven works, just updating system property at this point is too late
@@ -234,8 +233,7 @@
             && (!mojo.is("com.jayway.maven.plugins.android.generation2", "maven-android-plugin", "internal-integration-test"))
             && (!mojo.is("com.jayway.maven.plugins.android.generation2", "android-maven-plugin", "internal-integration-test"))
             && (!mojo.is("org.apache.maven.plugins", "maven-surefire-plugin", "test"))
-            && (!mojo.is("org.apache.maven.plugins", "maven-failsafe-plugin", "integration-test"))
-            && (!mojo.is("eviware", "maven-soapui-plugin", "test")))
+            && (!mojo.is("org.apache.maven.plugins", "maven-failsafe-plugin", "integration-test")))
             return false;
 
         try {
@@ -301,11 +299,6 @@
             } else if (mojo.is("org.codehaus.mojo", "gwt-maven-plugin", "test") && mojo.pluginName.version.compareTo("1.2") < 0) {
                     // gwt-maven-plugin < 1.2 does not implement required Surefire option
                     return false;
-            } else if (mojo.is("eviware", "maven-soapui-plugin", "test")) {
-                    Boolean skipTests = mojo.getConfigurationValue("skip", Boolean.class);
-                    if (((skipTests != null) && (skipTests))) {
-                        return false;
-                    }
             }
         } catch (ComponentConfigurationException e) {
             return false;
