From 0356493eee0d8d75c354057a49847f556ecc1e65 Mon Sep 17 00:00:00 2001
From: anb0s <anb0s@freenet.de>
Date: Wed, 19 Jan 2011 22:32:04 +0100
Subject: [PATCH] HUDSON-8497: Ability to use job parameter in all configuration fields (i.e. load rules)

---
 .../plugins/clearcase/AbstractClearCaseScm.java    |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/main/java/hudson/plugins/clearcase/AbstractClearCaseScm.java b/src/main/java/hudson/plugins/clearcase/AbstractClearCaseScm.java
index 7a607f1..79638c0 100644
--- a/src/main/java/hudson/plugins/clearcase/AbstractClearCaseScm.java
+++ b/src/main/java/hudson/plugins/clearcase/AbstractClearCaseScm.java
@@ -211,7 +211,7 @@ public abstract class AbstractClearCaseScm extends SCM {
 
     /**
      * Return string array containing the paths in the view that should be used when polling for changes.
-     * @param variableResolver TODO
+     * @param variableResolver [anb0s: HUDSON-8497]
      * @param build TODO
      * @param launcher TODO
      * @return string array that will be used by the lshistory command and for constructing the config spec, etc.
@@ -219,11 +219,14 @@ public abstract class AbstractClearCaseScm extends SCM {
      * @throws IOException 
      */
     public String[] getViewPaths(VariableResolver<String> variableResolver, AbstractBuild build, Launcher launcher) throws IOException, InterruptedException {
+    	// [--> anb0s: HUDSON-8497] 
         String loadRules = getLoadRules();
         if (StringUtils.isBlank(loadRules)) {
             return null;
         }
-
+        loadRules = Util.replaceMacro(loadRules, variableResolver);
+        // [<-- anb0s: HUDSON-8497]
+        
         String[] rules = loadRules.split("[\\r\\n]+");
         for (int i = 0; i < rules.length; i++) {
             String rule = rules[i];
-- 
1.7.3.1.msysgit.0

