### Eclipse Workspace Patch 1.0
#P cobertura
Index: src/main/resources/hudson/plugins/cobertura/dashboard/CoverageTablePortlet/main.jelly
===================================================================
--- src/main/resources/hudson/plugins/cobertura/dashboard/CoverageTablePortlet/main.jelly   (revision 34262)
+++ src/main/resources/hudson/plugins/cobertura/dashboard/CoverageTablePortlet/main.jelly   (working copy)
@@ -59,7 +59,7 @@
                           </j:choose>
                        </tr>
                      </j:forEach>
-                       <tr>
+                       <tr class="sortbottom">
                         <j:set var="result" value="${it.getCoverageResult(coverageRuns[0])}"/>
                         <j:set var="metrics" value="${result.metrics}"/>
                         <td><b>Total</b></td>
