diff --git a/src/main/java/hudson/plugins/jira/versionparameter/JiraVersionParameterDefinition.java b/src/main/java/hudson/plugins/jira/versionparameter/JiraVersionParameterDefinition.java
index 470596b..c63b485 100644
--- a/src/main/java/hudson/plugins/jira/versionparameter/JiraVersionParameterDefinition.java
+++ b/src/main/java/hudson/plugins/jira/versionparameter/JiraVersionParameterDefinition.java
@@ -18,14 +18,12 @@
 
 import hudson.EnvVars;
 import hudson.Extension;
-import hudson.model.AbstractBuild;
-import hudson.model.AbstractProject;
-import hudson.model.ParameterDefinition;
-import hudson.model.EnvironmentContributingAction;
-import hudson.model.ParameterValue;
+import hudson.model.*;
 import hudson.plugins.jira.JiraSession;
 import hudson.plugins.jira.JiraSite;
 import hudson.plugins.jira.soap.RemoteVersion;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 public class JiraVersionParameterDefinition extends ParameterDefinition {
 	private static final long serialVersionUID = 3927562542249244416L;
@@ -60,9 +58,30 @@
 		return value;
 	}
 
-	public List<JiraVersionParameterDefinition.Result> getVersions() throws IOException, ServiceException {
-		AbstractProject<?, ?> context = Stapler.getCurrentRequest().findAncestorObject(AbstractProject.class);
+   @Override
+   public ParameterValue getDefaultParameterValue() {
+      try {
+         if (!getVersions().isEmpty()) {
+            return new JiraVersionParameterValue(getName(), getVersions().get(0).name);
+         }
+      } catch (IOException ex) {
+         Logger.getLogger(JiraVersionParameterDefinition.class.getName()).log(Level.SEVERE, null, ex);
+      } catch (ServiceException ex) {
+         Logger.getLogger(JiraVersionParameterDefinition.class.getName()).log(Level.SEVERE, null, ex);
+      }
+      return null;
+   }
 
+   public List<JiraVersionParameterDefinition.Result> getVersions() throws IOException, ServiceException {
+      AbstractProject<?, ?> context = null;
+      if (Stapler.getCurrentRequest() != null) {
+         context = Stapler.getCurrentRequest().findAncestorObject(AbstractProject.class);
+      } else {
+         final List<AbstractProject> items = Hudson.getInstance().getItems(AbstractProject.class);
+         if (!items.isEmpty()) {
+            context = items.get(0);
+         }
+      }
 		JiraSite site = JiraSite.get(context);
         if (site==null)  throw new IllegalStateException("JIRA site needs to be configured in the project "+context.getFullDisplayName());
 
