Index: src/main/java/hudson/plugins/msbuild/MsBuildBuilder.java
===================================================================
--- src/main/java/hudson/plugins/msbuild/MsBuildBuilder.java	(revision 12073)
+++ src/main/java/hudson/plugins/msbuild/MsBuildBuilder.java	(working copy)
@@ -88,6 +88,8 @@
         if(normalizedTarget.trim().length()>0)
         	args.addTokenized(normalizedTarget);
         
+        args.addKeyValuePairs("-P:",build.getBuildVariables());
+        
         //If a msbuild file is specified, then add it as an argument, otherwise
         //msbuild will search for any file that ends in .proj or .sln
         if(msBuildFile != null && msBuildFile.trim().length() > 0){
