Index: src/main/java/hudson/maven/reporters/SurefireArchiver.java
===================================================================
--- src/main/java/hudson/maven/reporters/SurefireArchiver.java  (revision 37479)
+++ src/main/java/hudson/maven/reporters/SurefireArchiver.java  (working copy)
@@ -75,7 +75,7 @@
    }
     public boolean postExecute(MavenBuildProxy build, MavenProject pom, MojoInfo mojo, final BuildListener listener, Throwable error) throws InterruptedException, IOException {
-        if (!isSurefireTest(mojo)) return true;
+        //if (!isSurefireTest(mojo)) return true;
         listener.getLogger().println(Messages.SurefireArchiver_Recording());
