Index: src/main/java/hudson/plugins/performance/PerformanceReportMap.java
===================================================================
--- src/main/java/hudson/plugins/performance/PerformanceReportMap.java  (revision 32444)
+++ src/main/java/hudson/plugins/performance/PerformanceReportMap.java  (working copy)
@@ -52,24 +52,31 @@
         File repo = new File(getBuild().getRootDir(), PerformanceReportMap.getPerformanceReportDirRelativePath());
 
         // files directly under the directory is for JMeter, for a compatibility reasons.
-        List<File> pFileList = Arrays.asList(repo.listFiles(new FileFilter() {
+        // Note that this may fail, if the build itself filed, and we need to recover gracefully
+        File[] files = repo.listFiles(new FileFilter() {
             public boolean accept(File f) {
                 return !f.isDirectory();
             }
-        }));
-        addAll(new JMeterParser("").parse(getBuild(),pFileList,listener));
+        });
+        if (files != null) {
+            addAll(new JMeterParser("").parse(getBuild(),Arrays.asList(files),listener));
+        }
 
         // otherwise subdirectory name designates the parser ID.
-        for (File dir : repo.listFiles(new FileFilter() {
+        // Note that this may fail, if the build itself filed, and we need to recover gracefully
+        File[] dirs = repo.listFiles(new FileFilter() {
             public boolean accept(File f) {
                 return f.isDirectory();
             }
-        })) {
-            PerformanceReportParser p = buildAction.getParserById(dir.getName());
-            if (p!=null)
-                addAll(p.parse(getBuild(),Arrays.asList(dir.listFiles()),listener));
+        });
+        if (dirs != null) {
+            for (File dir : dirs) {
+                PerformanceReportParser p = buildAction.getParserById(dir.getName());
+                if (p!=null)
+                    addAll(p.parse(getBuild(),Arrays.asList(dir.listFiles()),listener));
+            }
         }
-       }
+    }
 
     private void addAll(Collection<PerformanceReport> reports) {
         for (PerformanceReport r : reports) {
