Index: src/test/java/hudson/plugins/sshslaves/SSHLauncherTest.java
===================================================================
--- src/test/java/hudson/plugins/sshslaves/SSHLauncherTest.java	(revision 0)
+++ src/test/java/hudson/plugins/sshslaves/SSHLauncherTest.java	(revision 0)
@@ -0,0 +1,64 @@
+package hudson.plugins.sshslaves;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.StringWriter;
+
+import junit.framework.TestCase;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class SSHLauncherTest extends TestCase {
+
+	@Test
+	public void testCheckJavaVersionOpenJDK7NetBSD() {
+		Assert.assertTrue("OpenJDK7 on NetBSD should be supported", checkSupported("openjdk-7-netbsd.version"));
+	}
+
+	@Test
+	public void testCheckJavaVersionOpenJDK6Linux() {
+		Assert.assertTrue("OpenJDK6 on Linux should be supported", checkSupported("openjdk-6-linux.version"));
+	}
+
+	@Test
+	public void testCheckJavaVersionSun6Linux() {
+		Assert.assertTrue("Sun 6 on Linux should be supported", checkSupported("sun-java-1.6-linux.version"));
+	}
+
+	@Test
+	public void testCheckJavaVersionSun6Mac() {
+		Assert.assertTrue("Sun 6 on Mac should be supported", checkSupported("sun-java-1.6-mac.version"));
+	}
+
+	@Test
+	public void testCheckJavaVersionSun4Linux() {
+		Assert.assertFalse("Sun 1.4 on Linux should NOT be supported", checkSupported("sun-java-1.4-linux.version"));
+	}
+
+	/**
+	 * Returns true if the version is supported.
+	 * 
+	 * @param testVersionOutput
+	 *            the resource to find relative to this class that contains the
+	 *            output of "java -version"
+	 * @return
+	 */
+	private static boolean checkSupported(final String testVersionOutput) {
+		try {
+			final String javaCommand = "testing-java";
+			final InputStream versionStream = SSHLauncherTest.class
+					.getResourceAsStream(testVersionOutput);
+			final BufferedReader r = new BufferedReader(new InputStreamReader(
+					versionStream));
+			final StringWriter output = new StringWriter();
+			final String result = SSHLauncher.checkJavaVersion(System.out,
+					javaCommand, r, output);
+			return null != result;
+		} catch (final IOException e) {
+			return false;
+		}
+	}
+}

Property changes on: src/test/java/hudson/plugins/sshslaves/SSHLauncherTest.java
___________________________________________________________________
Added: svn:mime-type
   + text/plain
Added: svn:keywords
   + Author Date Rev LastChangedBy Id
Added: svn:eol-style
   + native

Index: src/test/resources/hudson/plugins/sshslaves/openjdk-7-netbsd.version
===================================================================
--- src/test/resources/hudson/plugins/sshslaves/openjdk-7-netbsd.version	(revision 0)
+++ src/test/resources/hudson/plugins/sshslaves/openjdk-7-netbsd.version	(revision 0)
@@ -0,0 +1,3 @@
+openjdk version "1.7.0-internal"
+OpenJDK Runtime Environment (build 1.7.0-internal-pkgsrc_2010_01_03_06_54-b00)
+OpenJDK 64-Bit Server VM (build 17.0-b04, mixed mode)
Index: src/test/resources/hudson/plugins/sshslaves/sun-java-1.4-linux.version
===================================================================
--- src/test/resources/hudson/plugins/sshslaves/sun-java-1.4-linux.version	(revision 0)
+++ src/test/resources/hudson/plugins/sshslaves/sun-java-1.4-linux.version	(revision 0)
@@ -0,0 +1,3 @@
+java version "1.4.0"
+Java(TM) 2 Runtime Environment, Standard Edition (build 1.4.0-b92)
+Java HotSpot(TM) Client VM (build 1.4.0-b92, mixed mode)
Index: src/test/resources/hudson/plugins/sshslaves/sun-java-1.6-linux.version
===================================================================
--- src/test/resources/hudson/plugins/sshslaves/sun-java-1.6-linux.version	(revision 0)
+++ src/test/resources/hudson/plugins/sshslaves/sun-java-1.6-linux.version	(revision 0)
@@ -0,0 +1,3 @@
+java version "1.6.0_18"
+Java(TM) SE Runtime Environment (build 1.6.0_18-b07)
+Java HotSpot(TM) 64-Bit Server VM (build 16.0-b13, mixed mode)
Index: src/test/resources/hudson/plugins/sshslaves/sun-java-1.6-mac.version
===================================================================
--- src/test/resources/hudson/plugins/sshslaves/sun-java-1.6-mac.version	(revision 0)
+++ src/test/resources/hudson/plugins/sshslaves/sun-java-1.6-mac.version	(revision 0)
@@ -0,0 +1,3 @@
+java version "1.6.0_17"
+Java(TM) SE Runtime Environment (build 1.6.0_17-b04-248-10M3025)
+Java HotSpot(TM) 64-Bit Server VM (build 14.3-b01-101, mixed mode)
Index: src/test/resources/hudson/plugins/sshslaves/openjdk-6-linux.version
===================================================================
--- src/test/resources/hudson/plugins/sshslaves/openjdk-6-linux.version	(revision 0)
+++ src/test/resources/hudson/plugins/sshslaves/openjdk-6-linux.version	(revision 0)
@@ -0,0 +1,3 @@
+java version "1.6.0_0"
+OpenJDK Runtime Environment (IcedTea6 1.6.2) (suse-5.10.1-x86_64)
+OpenJDK 64-Bit Server VM (build 14.0-b16, mixed mode)
Index: src/main/java/hudson/plugins/sshslaves/SSHLauncher.java
===================================================================
--- src/main/java/hudson/plugins/sshslaves/SSHLauncher.java	(revision 29067)
+++ src/main/java/hudson/plugins/sshslaves/SSHLauncher.java	(working copy)
@@ -1,48 +1,53 @@
 package hudson.plugins.sshslaves;
 
+import static hudson.Util.fixEmpty;
+import static java.util.logging.Level.FINE;
+import hudson.AbortException;
+import hudson.Extension;
+import hudson.Util;
+import hudson.model.Descriptor;
+import hudson.model.Hudson;
+import hudson.model.TaskListener;
+import hudson.remoting.Channel;
+import hudson.slaves.ComputerLauncher;
+import hudson.slaves.SlaveComputer;
+import hudson.tools.JDKInstaller;
+import hudson.tools.JDKInstaller.CPU;
+import hudson.tools.JDKInstaller.Platform;
+import hudson.util.IOException2;
+import hudson.util.Secret;
+import hudson.util.StreamCopyThread;
+import hudson.util.StreamTaskListener;
+
+import java.io.BufferedOutputStream;
 import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
 import java.io.File;
 import java.io.IOException;
 import java.io.InputStreamReader;
+import java.io.PrintStream;
 import java.io.StringWriter;
-import java.io.ByteArrayOutputStream;
-import java.io.ByteArrayInputStream;
-import java.io.BufferedOutputStream;
+import java.net.URL;
+import java.text.NumberFormat;
+import java.text.ParseException;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Date;
 import java.util.List;
-import java.util.ArrayList;
+import java.util.logging.Logger;
 import java.util.regex.Pattern;
-import java.util.logging.Logger;
-import static java.util.logging.Level.FINE;
-import java.net.URL;
 
+import org.apache.commons.io.output.CountingOutputStream;
+import org.apache.commons.io.output.TeeOutputStream;
+import org.kohsuke.putty.PuTTYKey;
+import org.kohsuke.stapler.DataBoundConstructor;
+
 import com.trilead.ssh2.Connection;
 import com.trilead.ssh2.SFTPv3Client;
 import com.trilead.ssh2.SFTPv3FileAttributes;
 import com.trilead.ssh2.Session;
 import com.trilead.ssh2.StreamGobbler;
-import hudson.model.Descriptor;
-import hudson.model.Hudson;
-import hudson.model.TaskListener;
-import hudson.remoting.Channel;
-import hudson.slaves.ComputerLauncher;
-import hudson.slaves.SlaveComputer;
-import hudson.util.IOException2;
-import hudson.util.Secret;
-import hudson.util.StreamCopyThread;
-import hudson.util.StreamTaskListener;
-import hudson.Extension;
-import hudson.AbortException;
-import hudson.Util;
-import hudson.tools.JDKInstaller;
-import hudson.tools.JDKInstaller.Platform;
-import hudson.tools.JDKInstaller.CPU;
-import static hudson.Util.fixEmpty;
-import org.kohsuke.stapler.DataBoundConstructor;
-import org.kohsuke.putty.PuTTYKey;
-import org.apache.commons.io.output.TeeOutputStream;
-import org.apache.commons.io.output.CountingOutputStream;
 
 /**
  * A computer launcher that tries to start a linux slave by opening an SSH connection and trying to find java.
@@ -396,28 +401,62 @@
         ByteArrayOutputStream out = new ByteArrayOutputStream();
         connection.exec(javaCommand + " "+jvmOptions + " -version",out);
         BufferedReader r = new BufferedReader(new InputStreamReader(new ByteArrayInputStream(out.toByteArray())));
-        while (null != (line = r.readLine())) {
-            output.write(line);
-            output.write("\n");
-            line = line.toLowerCase();
-            if (line.startsWith("java version \"") || line.startsWith("openjdk version \"")) {
-                line = line.substring(line.indexOf('\"') + 1, line.lastIndexOf('\"'));
-                listener.getLogger().println(Messages.SSHLauncher_JavaVersionResult(getTimestamp(), javaCommand, line));
+        final String result = checkJavaVersion(listener.getLogger(), javaCommand, r, output);
 
-                // TODO make this version check a bit less hacky
-                if (line.compareTo("1.5") < 0) {
-                    // TODO find a java that is at least 1.5
-                    throw new IOException(Messages.SSHLauncher_NoJavaFound(line));
-                }
-                return javaCommand;
-            }
+        if(null == result) {
+        	listener.getLogger().println(Messages.SSHLauncher_UknownJavaVersion(javaCommand));
+        	listener.getLogger().println(output);
+        	throw new IOException(Messages.SSHLauncher_UknownJavaVersion(javaCommand));
+        } else {
+        	return result;
         }
-
-        listener.getLogger().println(Messages.SSHLauncher_UknownJavaVersion(javaCommand));
-        listener.getLogger().println(output);
-        throw new IOException(Messages.SSHLauncher_UknownJavaVersion(javaCommand));
     }
 
+	/**
+	 * Given the output of "java -version" in <code>r</code>, determine if this
+	 * version of Java is supported. This method has default visiblity for testing.
+	 * 
+	 * @param logger
+	 *            where to log the output
+	 * @param javaCommand
+	 *            the command executed, used for logging
+	 * @param r
+	 *            the output of "java -version"
+	 * @param out
+	 *            copy the data from <code>r</code> into this output buffer
+	 */
+	static String checkJavaVersion(final PrintStream logger, String javaCommand,
+			final BufferedReader r, final StringWriter output)
+			throws IOException {
+		String line;
+		while (null != (line = r.readLine())) {
+			output.write(line);
+			output.write("\n");
+			line = line.toLowerCase();
+			if (line.startsWith("java version \"")
+					|| line.startsWith("openjdk version \"")) {
+				final String versionStr = line.substring(
+						line.indexOf('\"') + 1, line.lastIndexOf('\"'));
+				logger.println(Messages.SSHLauncher_JavaVersionResult(
+						getTimestamp(), javaCommand, versionStr));
+
+				// parse as a number and we should be OK as all we care about is up through the first dot.
+				try {
+					final Number version =
+						NumberFormat.getNumberInstance().parse(versionStr);
+					if(version.doubleValue() < 1.5) {
+						throw new IOException(Messages
+								.SSHLauncher_NoJavaFound(line));
+					}
+				} catch(final ParseException e) {
+					throw new IOException(Messages.SSHLauncher_NoJavaFound(line));
+				}
+				return javaCommand;
+			}
+		}
+		return null;
+	}
+
     private void openConnection(TaskListener listener) throws IOException {
         listener.getLogger().println(Messages.SSHLauncher_OpeningSSHConnection(getTimestamp(), host + ":" + port));
         connection.connect();
